ocrd/ocrd/resolver.py

```diff
@@ -201,12 +216,21 @@ def workspace_from_nothing(self, directory, mets_basename='mets.xml', clobber_me
 
         return Workspace(self, directory, mets, mets_basename=mets_basename)
 
-    def resolve_mets_arguments(self, directory, mets_url, mets_basename):
+    def resolve_mets_arguments(self, directory, mets_url, mets_basename, mets_server_host, mets_server_port, mets_server_socket):
         """
-        Resolve the ``--mets``, ``--mets-basename`` and `--directory`` argument
+        Resolve the ``--mets``, ``--mets-basename``, `--directory``, ``--mets-server-host``,
+        ``--mets-server-port``, and ``--mets-server-socket`` arguments
         into a coherent set of arguments according to https://github.com/OCR-D/core/issues/517
         """
         log = getLogger('ocrd.resolver.resolve_mets_arguments')
+
+        # Determine --mets-server-*
+        if (mets_server_host or mets_server_port or mets_server_socket):
+            if mets_server_socket and (mets_server_host or mets_server_port):
+                raise ValueError('--mets-server-socket incompatible with --mets-server-host/--mets-server--port')
+            if (mets_server_host and not mets_server_port) or (not mets_server_host and mets_server_port):
```

Comment:
Since you like this kind of sorcery :), here is a shorter way to achieve the same:

```python
bool(mets_server_host) != bool(mets_server_port)
```

Reply:
🧙 Sorcery is appreciated, that's true, I am not sure whether this is more readable though. But it's clear from the context I think. I'll add it. 4f726ab64